### PR TITLE
Add `GenericInterfacesHelper.ExtractGenericInterface()`

### PR DESCRIPTION
**Pull request by dougbu**
nit: reorder test dependencies alphabetically

### CODE REVIEW

**Comment by dougbu** [comment]
This is part of the work on aspnet/Razor#89 (for realz, not the design PR).

**Comment by dougbu** [comment]
Updated: changed class name

File: src/Microsoft.Framework.ClosedGenericMatcher.Sources/ClosedGenericMatcher.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        /// <param name="queryType">The <see cref="Type"/> of interest.</param>
+        /// <param name="interfaceType">The open generic <see cref="Type"/> to match. Usually an interface.</param>
+        /// <returns>
+        /// The closed generic <see cref="Type"/> created from <paramref name="interfaceType"/> that
+        /// <paramref name="queryType"/> is or implements. <c>null</c> if the two <see cref="Type"/>s have no such
+        /// relationship.
+        /// </returns>
+        /// <remarks>
+        /// Name could have been <c>ExtractClosedGenericTypeMatchingOpenGenericInterfaceOrSealedType()</c> but that is
+        /// too long. In any case, this method will return <paramref name="queryType"/> if
```
Comment (rynowak) [inline src/Microsoft.Framework.ClosedGenericMatcher.Sources/ClosedGenericMatcher.cs]: The comments about the name don't really add anything, and won't be useful to readers 6 months from now. Just seems like snark :laughing:

**Comment by rynowak** [comment]
:shipit:

**Comment by dougbu** [comment]
aae8e6e

**Pull request merged**